app/routes.py

```diff
@@ -1 +1,235 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, jsonify, request, abort, make_response
+from app import db, Key
+from app.models.task import Task
+from app.models.goal import Goal
+import datetime
+import requests
+
+
+task_bp = Blueprint("task_bp", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goal_bp", __name__, url_prefix="/goals")
+
+#helper function to validate that object ids are valid
+def validate_id(cls, obj_id):
+    try:
+        obj_id = int(obj_id)
+    except ValueError:
+        response_str = f"Invalid {cls.__name__}_id: {obj_id} must be an integer"
+        abort(make_response(jsonify({"message":response_str}), 400))
+
+    matching_obj = cls.query.get(obj_id)
+
+    if not matching_obj:
+        response_str = f"Could not find {cls.__name__} with id {obj_id}"
+        abort(make_response(jsonify({"message":response_str}), 404))
+    
+    return matching_obj
+
+
+@task_bp.route("", methods=["POST"])
+def add_task():
+    request_body = request.get_json()
+    
+    if "title" not in request_body or "description" not in request_body:
+        abort(make_response(jsonify({"details":"Invalid data"}), 400))
```

Comment:
It's good that you're doing input validation here, but maybe also let the end-user know what/why the data is invalid.